### PR TITLE
Integrated Experiment Revisions & Bug fixes

### LINKED ISSUE #1009: Integrated Learner doesn't save state when processing test curriculum
The learner doesn't save a state as the test curriculum is running however the observers do record themselves.

### PR DESCRIPTION
**Pull request by LichMaster98**
Closes #1009

### CODE REVIEW

File: adam/experiment/README.md [hunk 1]
```diff
@@ -97,6 +98,10 @@ None
 - smoothing_parameter: Float - A value between 0 and 1 which is generally small which smooths the probabilities over time.
 - random_seed: Integer (Optional: 0) - A seed for the RandomChooser which is used in this learner
 
+### Object Recognizer
+*Valid For: Object*
+While not strictly a 'learner', this option builds in the knowledge of objects to the integrated learner removing any doubt around the learner's ability to recognize objects.
+
```
Comment (spigo900) [inline adam/experiment/README.md:104]: Good thought documenting this. `suppress_error` too.

**Review by spigo900 — APPROVED** [review]
Looks good, just two minor thoughts

File: adam/experiment/log_experiment.py [hunk 2]
```diff
@@ -653,7 +655,12 @@ def curriculum_from_params(
                 language_generator,
                 params=params.namespace_or_empty("train_curriculum"),
             ),
-            test_instance_groups(num_samples, num_noise_objects, language_generator)
+            test_instance_groups(
+                5,
+                0,
```
[the remark this answers is not in the document]
Comment (LichMaster98) [reply adam/experiment/log_experiment.py:660]: So yes, we repeat the number of instances and number of noise instances that go into both the training and test curriculum on one parameter. Given I don't think time allows for refactoring all the curriculum to ensure that separating these values into two parameters I've just hard coded the specific number of test examples for the two curricula in this branch to what they should be. -- Probably should have tracked this with a git issue?

File: adam/curriculum/curriculum_utils.py [hunk 3]
```diff
@@ -168,8 +178,97 @@ def learner_template_factory() -> TemplateObjectVariable:
...
-) -> Sequence[Phase1InstanceGroup]:
-    mod_curriculum = list(flatten(curriculum))
+    curriculum: List[
+        Tuple[SituationT, LinguisticDescriptionT, PerceptualRepresentation[PerceptionT]]
+    ],
+    *,
+    rng: Random,
+) -> Sequence[
+    Tuple[SituationT, LinguisticDescriptionT, PerceptualRepresentation[PerceptionT]]
+]:
```
[the remark this answers is not in the document]
Comment (LichMaster98) [reply adam/curriculum/curriculum_utils.py:188]: No actually this fixes the flawed implementation :) (There was a bug related to the noise generation that was previously hiding that we weren't shuffling the examples)

File: adam/experiment/log_experiment.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (spigo900) [reply adam/experiment/log_experiment.py:660]: No, that's fine. I was just checking if this was intentional, and it sounds like it is.

File: adam/curriculum/curriculum_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (spigo900) [reply adam/curriculum/curriculum_utils.py:188]: Ah, okay, sounds good!

**Pull request merged**